kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -348,22 +348,19 @@ class ResidualBasedBlockBuilderAndSolver
         else
             norm_b = 0.00;
 
-        if (norm_b != 0.00)
-        {
+        if (norm_b != 0.00) {
             //provide physical data as needed
             if(BaseType::mpLinearSystemSolver->AdditionalPhysicalDataIsNeeded() )
                 BaseType::mpLinearSystemSolver->ProvideAdditionalData(A, Dx, b, BaseType::mDofSet, rModelPart);
 
             //do solve
             BaseType::mpLinearSystemSolver->Solve(A, Dx, b);
-        }
-        else
-        {
+        } else {
             TSparseSpace::SetToZero(Dx);
-            KRATOS_WARNING("ResidualBasedBlockBuilderAndSolver") << "ATTENTION! setting the RHS to zero!" << std::endl;
+            KRATOS_WARNING_IF("ResidualBasedBlockBuilderAndSolver", this->GetEchoLevel() > 0) << "ATTENTION! setting the RHS to zero!" << std::endl;

```

Comment:
@KratosMultiphysics/technical-committee I added this check for echo level, please tell me if you agree, otherwise I will revert. But due to many patch tests we have the number of tests with zero RHS fills the screen with this warning

Reply:
I definitively keep this warning as it is.

Reply:
Al alternative I was thinking is to define the echo level as -1 for tests (it is an integer, so is possible). Otherwise it give many warnings that are not use. When solving certain problems like patch tests is normal to have a zero norm RHS

Reply:
@loumalouomega this modification does not belong here. plese leave this file as it was...

Reply:
OK, any idea how to avoid this messages (when it is intentioned)

Reply:
@loumalouomega i don't have an alternative proposal. we can discuss what should be done in a different PR, however it does not belong here